.github/workflows/main.yml

```diff
@@ -20,7 +20,7 @@ jobs:
   build:
     strategy:
       matrix:
-        name: ["Ubuntu Latest GCC", "macOS Latest Clang"]
+        name: ["Ubuntu Latest GCC", "macOS Latest"]
```

Comment:
Maybe you should use macos-13, which is x86, so we can still test Fortran on Macs.

Reply:
Indeed a separate runner is needed for x86 Mac. But my goal is to achieve M1/M2/M3 compatibility with Fortran enabled. I have deactivated temporary Fortran and will try to solve the issue as soon as possible. So far, I suspect that transfer of strings between C and Fortran on arm64 may be part of the problem... and I have not figured out why.
I will try to enable an interactive github action to investigate if possible.

Reply:
Ok, thanks. I did not realize you were still looking into it. I think flang is now being used, which could be part of the issue. flang was at least an issue for HDF5 Fortran on macs in general.

Reply:
@brtnfld Problem solved \o/

The solution works with clang and gcc.